homeassistant/components/esphome/__init__.py

```diff
@@ -325,6 +325,7 @@ def async_on_state_subscription(entity_id: str) -> None:
             # In the future another API will be set up so that the ESP can
             # notify HA of connectivity directly, but for new we'll use a
             # really short reconnect interval.
+            tries = min(tries, 10)  # prevent OverflowError
             wait_time = int(round(min(1.8**tries, 60.0)))
```

Comment:
The fix seems good but I wonder if this could be rewritten to look less like magic. Or maybe there is indeed some magic reason for `1.8`?

Reply:
It was 2 first, but that grew too fast. So I tried different values and 1.8 was the most reasonable. Nothing magic here ;)

Reply:
If you are experimenting for reasonable values, maybe a static list is better?
```
wait_time = ([1, 3, 5, 10, 20, 30][tries-1:] or [60])[0]
```
Now, maybe _that_ looks like magic ...

Reply:
I mean that would be possible but the current system works fine as-is - the exact timing isn't important. Just that the first disconnect gets handled quickly and later on you don't churn away resources.

I don't think it's of importance how this logic is implemented tbh. Plus exponential backoff is standard for this problem.

Reply:
Well, according to this PR it didn't exactly work fine ;-)

Reply:
True :)